packages/admin/cms-admin/src/dam/DataGrid/copyPaste/useCopyPasteDamItems.tsx

```diff
@@ -0,0 +1,104 @@
+}
+
+const copyFilesMutation = gql`
+    mutation CopyPasteFiles($fileIds: [ID!]!, $targetFolderId: ID, $targetScope: DamScopeInput) {
+        copyDamFiles(fileIds: $fileIds, targetFolderId: $targetFolderId, targetScope: $targetScope) {
+            mappedFiles {
+                rootFile {
+                    id
+                }
+                copy {
+                    id
+                }
+            }
+        }
+    }
+`;
+
+export const useCopyPasteDamItems = () => {
+    const currScope = useDamScope();
+    const apolloClient = useApolloClient();
+
+    const prepareForClipboard = React.useCallback((damItems: Array<{ id: string; type: "file" | "folder" }>): DamItemsClipboard => {
+        return { damItems };
+    }, []);
+
+    const writeToClipboard = React.useCallback((items: DamItemsClipboard) => {
+        return writeClipboardText(JSON.stringify(items));
+    }, []);
+
+    const getFromClipboard = React.useCallback(async (): Promise<GetFromClipboardResponse> => {
+        const text = await readClipboardText();
+
+        if (text === undefined) {
+            return {
+                canPaste: false,
+                error: <FormattedMessage {...messages.failedToReadClipboard} />,
+            };
+        }
+
+        if (text.trim() === "") {
+            return {
+                canPaste: false,
+                error: <FormattedMessage {...messages.emptyClipboard} />,
+            };
+        }
+
+        try {
+            const parsedText = JSON.parse(text);
+            if (isDamItemsClipboard(parsedText)) {
+                return { canPaste: true, content: parsedText };
+            } else {
+                throw new Error("Invalid clipboard content");
+            }
+        } catch {
+            return {
+                canPaste: false,
+                error: (
+                    <FormattedMessage
+                        id="comet.dam.cannotPasteFiles.messageFailedToParseClipboard"
+                        defaultMessage="Content from clipboard aren't valid DAM items"
+                    />
+                ),
+            };
+        }
+    }, []);
+
+    const createCopies = React.useCallback(
+        async ({ clipboard: { damItems }, targetFolderId }: { clipboard: DamItemsClipboard; targetFolderId: string | undefined }) => {
+            const fileIds = damItems.filter(({ type }) => type === "file").map((file) => file.id);
+
+            await apolloClient.mutate<GQLCopyPasteFilesMutation, GQLCopyPasteFilesMutationVariables>({
+                mutation: copyFilesMutation,
+                variables: { fileIds, targetFolderId, targetScope: currScope },
+                update: (cache) => {
+                    cache.evict({ fieldName: "damItemsList" });
+                },
+            });
+        },
+        [apolloClient, currScope],
+    );
+
+    return { prepareForClipboard, writeToClipboard, getFromClipboard, createCopies };
```

Comment:
Yes, I built this with regard to the `useCopyPastePages` API because I wanted to stay somewhat consistent. But I also don't particularly like the API.

I reworked it. Now it only exports `writeToClipboard ` and `pasteFromClipboard `. I kept `getFromClipboard` and `createCopies` as internal functions because I think the split makes sense and the functions are shorter.